packages/jaspr/lib/src/foundation/styles/properties/filter.dart

```diff
@@ -0,0 +1,188 @@
+import 'angle.dart';
+import 'color.dart';
+import 'unit.dart';
+
+typedef BackdropFilter = Filter;
+
+class Filter {
+  const Filter._(this.value);
+
+  /// The css value
+  final String value;
+
+  // Keyword value(s)
+  static const Filter none = Filter._('none');
+
+  // Global values
+  static const Filter inherit = Filter._('inherit');
+  static const Filter initial = Filter._('initial');
+  static const Filter revert = Filter._('revert');
+  static const Filter revertLayer = Filter._('revert-layer');
+  static const Filter unset = Filter._('unset');
+
+  const factory Filter(List<FilterFn> filterValues) = _Filter;
+}
+
+class _Filter implements Filter {
+  const _Filter(this._filterValues);
+
+  final List<FilterFn> _filterValues;
+
+  /// The css value
+  @override
+  String get value => _filterValues.map((e) => e.value).join(' ');
+}
+
+abstract class FilterFn {
+  const factory FilterFn.blur([Unit? length]) = _BlurFilter;
+  const factory FilterFn.brightness([double? percentage]) = _BrightnessFilter;
+  const factory FilterFn.contrast([double? percentage]) = _ContrastFilter;
+  const factory FilterFn.dropShadow({
+    required Unit offsetX,
+    required Unit offsetY,
+    Unit? spread,
+    Color? color,
+  }) = _DropShadowFilter;
+  const factory FilterFn.grayscale([double? percentage]) = _GrayscaleFilter;
+  const factory FilterFn.hueRotate([Angle? angle]) = _HueRotateFilter;
+  const factory FilterFn.invert([double? percentage]) = _InvertFilter;
+  const factory FilterFn.opacity([double? percentage]) = _OpacityFilter;
+  const factory FilterFn.sepia([double? percentage]) = _SepiaFilter;
+  const factory FilterFn.saturate([double? percentage]) = _SaturateFilter;
+  const factory FilterFn.url(String url) = _UrlFilter;
+  const factory FilterFn.src(String url) = _SrcFilter;
+
+  /// The css value
+  String get value;
+}
+
+class _BlurFilter implements FilterFn {
+  const _BlurFilter([this._length]);
+
+  final Unit? _length;
+
+  /// The css value
+  @override
+  String get value => "blur(${_length?.value ?? ''})";
+}
+
+class _BrightnessFilter implements FilterFn {
```

Comment:
You can add a shared superclass for these that are all some percentage in filter function.

e.g.
```dart
class _FilterWithPercentage implements Filter {
  _FilterWithPercentage(this._percentage, this._name);

  final double? _percentage;
  final String _name;

  @override
  String get value => "$_name(${_percentage?.numstr ?? ''})";
}
```

and then

```dart
class _BrightnessFilter extends _FilterWithPercentage {
  const _BrightnessFilter(super._percentage) : super('brightness');
}
```